pylint/checkers/classes.py

```diff
@@ -907,14 +907,25 @@ def _check_unused_private_members(self, node: astroid.ClassDef) -> None:
             found = False
             if not is_attr_private(function_def.name):
                 continue
-            for attribute in node.nodes_of_class(astroid.Attribute):
+            attribute: astroid.Attribute
+            for attribute in (
+                a
+                for a in node.nodes_of_class(astroid.Attribute)
+                if a.attrname == function_def.name
+                and a.scope() != function_def  # We ignore recursive calls
+                and a.expr is not None  # Probably not really an Optional ?
+            ):
```

Comment:
This is really complicated. It would be better to iterate over all and just do an `if ... continue` afterwards.